compose/src/main/java/com/bottlerocketstudios/compose/map/GoogleMapsView.kt

```diff
@@ -49,101 +49,141 @@ fun GoogleMapsView(googleMapScreenState: GoogleMapScreenState, toolbarEnabled: B
+                    onclick = { marker ->
+                        Timber.d("${marker.id} ${marker.title}")
+                        googleMapScreenState.setSelectedMarker(marker.tag as YelpMarker)
+                        googleCameraPositionState.move(
+                            CameraUpdateFactory.newLatLng(
+                                LatLng(
+                                    marker.position.latitude,
+                                    marker.position.longitude
+                                )
+                            )
+                        )
 
-        AnimatedVisibility(dialogVisibility.value) {
-            when (val error = googleMapScreenState.yelpError.value) {
-                is UserFacingError.ApiError -> CustomAlertDialog(
-                    modifier = Modifier,
-                    title = error.title,
-                    message = error.description,
-                    onDismiss = {
-                        googleMapScreenState.resetError()
-                    }
+                        true
+                    },
+                    googleMapScreenState.yelpMarkerSelected.value
                 )
-                is UserFacingError.GeneralError -> CustomAlertDialog(
-                    modifier = Modifier,
-                    title = error.title,
-                    message = error.description,
-                    onDismiss = {
-                        googleMapScreenState.resetError()
-                    }
+            }
+
+            if (googleCameraPositionState.isMoving) {
+                val search = YelpLatLngSearch(
+                    latitude = googleCameraPositionState.position.target.latitude,
+                    longitude = googleCameraPositionState.position.target.longitude
                 )
-                else -> {
-                    googleMapScreenState.resetError()
-                }
+                googleMapScreenState.onCameraMoveSearch(search, googleCameraPositionState.position.zoom)
             }
         }
 
-        Button(onClick = {
-            mapUiSettings = mapUiSettings.copy(
-                mapToolbarEnabled = !mapUiSettings.mapToolbarEnabled
+        AnimatedVisibility(dialogVisibility.value) {
+            ShowErrorDialog(
+                yelpError = googleMapScreenState.yelpError.value,
+                onDismiss = {
+                    googleMapScreenState.resetError()
+                    showRetryButton.value = true
+                }
             )
-        }) {
-            Text(text = stringResource(R.string.toggle_map_toolbar))
         }
 
         AnimatedVisibility(googleMapScreenState.businessList.value.isNotEmpty()) {
-            YelpBusinessList(businessList = googleMapScreenState.businessList.value)
+            YelpBusinessList(
+                businessList = googleMapScreenState.businessList.value, onClick = { yelpBusiness ->
+                    googleMapScreenState.setSelectedMarker(
```

Comment:
In a coming PR, we'll consolidate some of our logic, by delaying converting yelpBusiness to YelpMarker until very last second (or not at all).  This will make the need to wrap/unwrap go away and will tighten the code a bit more.